server/pulp/server/managers/schedule/repo.py

```diff
@@ -250,22 +241,15 @@ def delete_by_distributor_id(repo_id, distributor_id):
         """
         Delete all schedules for the specified repo and distributor.
 
-        :param distributor_id:  unique ID for an distributor
-        :type  distributor_id:  basestring
-        """
-        utils.delete_by_resource(RepoDistributor.build_resource_tag(repo_id, distributor_id))
-
-    @staticmethod
-    def validate_distributor(repo_id, distributor_id):
-        """
-        Validate that the distributor exists for the specified repo
-
-        :param repo_id:         unique ID for a repository
-        :type  repo_id:         basestring
+        :param repo_id:  unique ID for a repository
+        :type  repo_id:  basestring
         :param distributor_id:  unique ID for a distributor
         :type  distributor_id:  basestring
-
-        :raise: pulp.server.exceptions.MissingResource
         """
-        distributor_manager = managers_factory.repo_distributor_manager()
-        distributor_manager.get_distributor(repo_id, distributor_id)
+        try:
+            dist = model.Distributor.objects.get_or_404(repo_id=repo_id,
+                                                        distributor_id=distributor_id)
+        except exceptions.MissingResource:
+            pass
```

Comment:
I agree that deleting the schedules should be idempotent. Call this as many times as you want on a distributor, and we'll make sure that distributor doesn't have any schedules. But this function is not responsible for deleting the distributor itself, so it should complain if the distributor doesn't exist. If the caller happens to also be trying to delete the distributor, it can ignore that error if it wants.